### ISSUE TITLE
Allow dependencies between fields

### ISSUE THREAD

**Issue by socunanena** [eivindfjeldstad/validate#76]
It would be great allowing dependencies between two or more fields. For example, if I have this user:
```javascript
const user = new Schema({
  username: {
    type: String,
    required: true,
    length: { min: 3, max: 32 }
  },
  email: {
    type: String
  },
  phoneNumber: {
    type: String
  }
})
```
Validate that this user has registered its email _or_ its phone number. That is, if the email is provided, I don't need (or I don't want) the phone number, and viceversa. Use cases:

- Require both fields (already implemented)
- Require **just one** field: if email is provided, **fail** when phoneNumber (or other dependencies) is provided. Fail when no dependencies are provided.
- Require **at least one** field: if email is provided, **don't fail** when phoneNumber (or other dependencies) is provided. Fail when no dependencies are provided.

Many thanks :)

**Comment by socunanena** [comment]
Yes, I tried something like that, but it doesn't really work since `validate/property.js` is validating the `required` type in the first place: if there's no required validation and no value, it stops validating the following validations, so it never reaches the `maybeRequired` one.

**Comment by eivindfjeldstad** [comment]
Ah sorry, of course you're right. That's actually a very persuasive argument for fixing #65

**Comment by eivindfjeldstad** [comment]
Should be fixed with commit d1c78c9
